api/src/services/items.ts

```diff
@@ -419,31 +428,30 @@ export class ItemsService<Item extends AnyItem = AnyItem> implements AbstractSer
 				  )
 				: query;
 
-		let ast = await getASTFromQuery(this.collection, updatedQuery, this.schema, {
-			accountability: this.accountability,
-			// By setting the permissions action, you can read items using the permissions for another
-			// operation's permissions. This is used to dynamically check if you have update/delete
-			// access to (a) certain item(s)
-			action: opts?.permissionsAction || 'read',
-			knex: this.knex,
-		});
-
-		if (this.accountability && this.accountability.admin !== true) {
-			const authorizationService = new AuthorizationService({
+		let ast = await getAstFromQuery(
+			{
+				collection: this.collection,
+				query: updatedQuery,
 				accountability: this.accountability,
-				knex: this.knex,
+			},
+			{
 				schema: this.schema,
-			});
+				knex: this.knex,
+			},
+		);
 
-			ast = await authorizationService.processAST(ast, opts?.permissionsAction);
-		}
+		ast = await processAst(
+			{ ast, action: 'read', accountability: this.accountability },
```

Comment:
Did we drop support for `QueryOptions.permissionsAction` or is this accounted for somewhere else, that I'm missing?